onnxruntime/core/mlas/lib/intrinsics/avx512/quantize_avx512f.cpp

```diff
@@ -0,0 +1,172 @@
+
+    while (N >= 64) {
+
+        auto FloatVector0 = _mm512_loadu_ps(Input);
+        auto FloatVector1 = _mm512_loadu_ps(Input + 16);
+        auto FloatVector2 = _mm512_loadu_ps(Input + 32);
+        auto FloatVector3 = _mm512_loadu_ps(Input + 48);
+
+        FloatVector0 = _mm512_div_ps(FloatVector0, ScaleVector);
+        FloatVector1 = _mm512_div_ps(FloatVector1, ScaleVector);
+        FloatVector2 = _mm512_div_ps(FloatVector2, ScaleVector);
+        FloatVector3 = _mm512_div_ps(FloatVector3, ScaleVector);
+
+        FloatVector0 = _mm512_max_ps(FloatVector0, MinimumValueVector);
+        FloatVector1 = _mm512_max_ps(FloatVector1, MinimumValueVector);
+        FloatVector2 = _mm512_max_ps(FloatVector2, MinimumValueVector);
+        FloatVector3 = _mm512_max_ps(FloatVector3, MinimumValueVector);
+
+        FloatVector0 = _mm512_min_ps(FloatVector0, MaximumValueVector);
+        FloatVector1 = _mm512_min_ps(FloatVector1, MaximumValueVector);
+        FloatVector2 = _mm512_min_ps(FloatVector2, MaximumValueVector);
+        FloatVector3 = _mm512_min_ps(FloatVector3, MaximumValueVector);
+
+        auto IntegerVector0 = _mm512_cvtps_epi32(FloatVector0);
+        auto IntegerVector1 = _mm512_cvtps_epi32(FloatVector1);
+        auto IntegerVector2 = _mm512_cvtps_epi32(FloatVector2);
+        auto IntegerVector3 = _mm512_cvtps_epi32(FloatVector3);
+
+        IntegerVector0 = _mm512_add_epi32(IntegerVector0, ZeroPointVector);
+        IntegerVector1 = _mm512_add_epi32(IntegerVector1, ZeroPointVector);
+        IntegerVector2 = _mm512_add_epi32(IntegerVector2, ZeroPointVector);
+        IntegerVector3 = _mm512_add_epi32(IntegerVector3, ZeroPointVector);
+
+        auto ByteVector0 = _mm512_cvtepi32_epi8(IntegerVector0);
+        auto ByteVector1 = _mm512_cvtepi32_epi8(IntegerVector1);
+        auto ByteVector2 = _mm512_cvtepi32_epi8(IntegerVector2);
+        auto ByteVector3 = _mm512_cvtepi32_epi8(IntegerVector3);
+        _mm_storeu_si128((__m128i*)Output, ByteVector0);
+        _mm_storeu_si128((__m128i*)(Output+16), ByteVector1);
+        _mm_storeu_si128((__m128i*)(Output+32), ByteVector2);
+        _mm_storeu_si128((__m128i*)(Output+48), ByteVector3);
+
+        Input += 64;
+        Output += 64;
+        N -= 64;
+    }
+
+    while (N >= 16) {
+        auto FloatVector = _mm512_loadu_ps(Input);
+        FloatVector = _mm512_div_ps(FloatVector, ScaleVector);
+        FloatVector = _mm512_max_ps(FloatVector, MinimumValueVector);
+        FloatVector = _mm512_min_ps(FloatVector, MaximumValueVector);
+
+        auto IntegerVector = _mm512_cvtps_epi32(FloatVector);
+        IntegerVector = _mm512_add_epi32(IntegerVector, ZeroPointVector);
+
+        auto ByteVector = _mm512_cvtepi32_epi8(IntegerVector);
+        _mm_storeu_si128((__m128i*)Output, ByteVector);
+
+        Input += 16;
+        Output += 16;
+        N -= 16;
+    }
+
+    MlasQuantizeLinearKernel(Input, Output, N, Scale, ZeroPoint);
```

Comment:
Calling the default version here for the remaining count worries me a little as the mixing of intrinsic code in a single binary can cause problem with LTCG (at least Windows). You could use the same logic that you have in the N>=16 case, but use _mm512_mask_loadu_ps and _mm_mask_storeu_ps to store the partial bytes, the rest stays the same. Use _cvtu32_mask16 to build the mask from the remaining N. I suspect the generated code for this function would then also be smaller as it would then be a leaf function.